framework/doc/content/source/userobject/RenormalizeVector.md

```diff
@@ -0,0 +1,21 @@
+# RenormalizeVector
+
+!syntax description /UserObjects/RenormalizeVector
+
+This user object can renormalize the solution for a set of variables, taking each variable as the component of a vector and scaling the variables to obtain the user specified L2-norm.
```

Comment:
add a note that this is a DOF vector norm not anything physical

Reply:
This is explained in the Design section.

Reply:
user should be able to understand what this does without knowing how it s designed. 
This doesnt convey the local character of the renormalization

Reply:
added "pointwise"

Reply:
RenormalizePointwiseVector?

Reply:
In the description. What's a "pointwise vector"?

Reply:
No idea. Let s do perDoF then instead of pointwise

Reply:
How would you even do the renormalization any other way?!

Reply:
Sum them all over all dofs, compare to a norm, renormalize.

Reply:
really one would want to use a volume integral, then renormalize the vector of variables, because that's more physical.

This needs an indication that it's a local operation, a normalization per DOF.
I m happy with Local or perDoF in the name. Feel free to find better

Reply:
> really one would want to use a volume integral, then renormalize the vector of variables, because that's more physical.

Integrate over which volume?

Reply:
> because that's more physical.

I feel this discussion has gone of the rails. How else would you explain that comment. "More physical" ... The issue mentions micro magnetics. Are you familiar with those physics? The solution is a field of unit vectors indicating a direction. Integrating over a volume does not make sense and certainly isn't "more physical".

Reply:
Integrate over the whole volume, like one does when we renormalize power density to conserve energy for example. My remarks about physicality are in the larger context of the name of this object, not in the local magnetics context. I dont contest that it's a great addition to be able to renormalize to a local unit sphere

The name ties it to other contexts, renormalization is not just a micro-magnetics field. This object is going in the framework not the EM module, but even then it shows up in the general index so we can't

Reply:
A wider issue we need an issue for is that we shouldnt need something as free-form as UOs to do this. There should be AuxKernels for nonlinear variables.